tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/oidc/OidcAuthenticationStrategy.java

```diff
@@ -0,0 +1,101 @@
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+
+package com.linagora.tmail.james.jmap.oidc;
+
+import static org.apache.james.jmap.http.JWTAuthenticationStrategy.AUTHORIZATION_HEADER_PREFIX;
+
+import java.time.Clock;
+
+import jakarta.inject.Inject;
+
+import org.apache.james.core.Username;
+import org.apache.james.jmap.exceptions.UnauthorizedException;
+import org.apache.james.jmap.http.AuthenticationChallenge;
+import org.apache.james.jmap.http.AuthenticationScheme;
+import org.apache.james.jmap.http.AuthenticationStrategy;
+import org.apache.james.jwt.introspection.TokenIntrospectionException;
+import org.apache.james.jwt.userinfo.UserInfoCheckException;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.SessionProvider;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.ImmutableMap;
+import com.linagora.tmail.james.jmap.model.Aud;
+import com.linagora.tmail.james.jmap.model.Token;
+import com.linagora.tmail.james.jmap.model.TokenInfo;
+
+import reactor.core.publisher.Mono;
+import reactor.netty.http.server.HttpServerRequest;
+
+public class OidcAuthenticationStrategy implements AuthenticationStrategy {
+    private final SessionProvider sessionProvider;
+    private final TokenInfoResolver tokenInfoResolver;
+    private final Clock clock;
+    private final Aud aud;
+
+    @Inject
+    public OidcAuthenticationStrategy(SessionProvider sessionProvider, TokenInfoResolver tokenInfoResolver, Clock clock, Aud aud) {
+        this.sessionProvider = sessionProvider;
+        this.tokenInfoResolver = tokenInfoResolver;
+        this.clock = clock;
+        this.aud = aud;
+    }
+
+
+    @Override
+    public Mono<MailboxSession> createMailboxSession(HttpServerRequest httpRequest) {
+        return Mono.fromCallable(() -> authHeaders(httpRequest))
+            .filter(header -> header.startsWith(AUTHORIZATION_HEADER_PREFIX))
+            .map(header -> header.substring(AUTHORIZATION_HEADER_PREFIX.length()))
+            .filter(token -> !token.startsWith("eyJ")) // Heuristic for detecting JWT
```

Comment:
This heuristic is used for Twake calendar because we NEED to mix legacy JWT with OIDC but I do not think it make sense to twake mail.

Worse, doing do forces a reconfiguration of lemonLDAP while doing the upgrade.

Let's drop it?